### PR TITLE
feat(jexl): add support for nested forms

### PR DESCRIPTION
**Pull request by czosel**
See https://github.com/projectcaluma/ember-caluma/pull/154 for the corresponding frontend implementation.

### CODE REVIEW

File: caluma/form/jexl.py [hunk 1]
```diff
@@ -15,8 +15,21 @@ class QuestionJexl(JEXL):
     def __init__(self, answer_by_question={}, **kwargs):
         super().__init__(**kwargs)
 
-        self.add_transform("answer", lambda question: answer_by_question.get(question))
+        self.context = answer_by_question
+        self.add_transform("answer", self.answer_transform)
         self.add_transform("mapby", lambda arr, key: [obj[key] for obj in arr])
 
+    def answer_transform(self, *args):
```
Comment (winged) [inline caluma/form/jexl.py]: I'd prefer explicit arguments - they have specific meanings anyway, judging from the code.

You can do `def answer_transform(self, question, path=None):` to get the same effect

File: caluma/form/tests/test_jexl.py [hunk 2]
```diff
@@ -19,3 +19,22 @@
 def test_question_jexl_validate(expression, num_errors):
     jexl = QuestionJexl()
     assert len(list(jexl.validate(expression))) == num_errors
+
+
+@pytest.mark.parametrize(
+    "expression,result",
+    [
+        ('"a1"|answer', "A1"),
```
Comment (winged) [inline caluma/form/tests/test_jexl.py]: That interface feels awkward. Why not unify the path with the question slug? Something like `"parent.form_b.b1"|answer`

File: caluma/form/jexl.py (hunk 1 above)
Comment (czosel) [reply caluma/form/jexl.py]: Right, that's what I was looking for! My python is a bit rusty :blush:

File: caluma/form/tests/test_jexl.py (hunk 2 above)
Comment (czosel) [reply caluma/form/tests/test_jexl.py]: Good point. 
/cc @fkm @kaldras

Comment (kaldras) [reply caluma/form/tests/test_jexl.py]: Yeah, I like this. Feels simpler and does the same!

Comment (czosel) [reply caluma/form/tests/test_jexl.py]: Fixed!

File: caluma/form/jexl.py (hunk 1 above)
Comment (czosel) [reply caluma/form/jexl.py]: Fixed

**Pull request merged**